### ISSUE TITLE
Add option to disable arrow-body-style rule when the function contains a long body

### ISSUE THREAD

**Issue by bendtherules** [eslint/eslint#10635]
**What rule do you want to change?**
arrow-body-style

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add option to "as-needed" value to ignore when the function body contains more than n number of lines.
I think its good practise to have curly braces if any arrow function gets bigger. Say more than 4 lines.

**Please provide some example code that this change will affect:**

```js
(something) => {
    return (
        <div>
            <i className="icon" />
            <div className="text">
                {something}
            </div>
            <div className="text">
                {something}
            </div>
        </div>
    );
}
```

**What does the rule currently do for this code?**
Shows warning/error

**What will the rule do after it's changed?**
Not show warning/error

**Comment by bendtherules** [comment]
Thanks. That also looks good.

Closing.
